src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -348,16 +377,25 @@ exports.createFiberFromText = function(
 function createFiberFromElementType(
   type: mixed,
   key: null | string,
+  internalContextTag: TypeOfInternalContext,
   debugOwner: null | Fiber | ReactInstance,
 ): Fiber {
+  if (
+    ReactFeatureFlags.enableAsyncSubtreeAPI &&
+    type != null &&
+    (type: any).unstable_asyncUpdates === true
```

Comment:
@acdlite This `any` cast is surfacing a flaw. This tests for `unstable_asyncUpdates` on strings. Which theoretically could exist (although very unlikely). More likely is that it will mess with some plausible optimization and type safety.

This should at the very least move into each block below.

Although it would be even better if it moved all the way into the updateClassComponent so that it is only available on true classes.

We have an unfortunate polymorphic check to enter classes but once we're in the class path everything is slow and polymorphic anyway. I'd prefer to keep all these weird flags in that path so that we can leave functional components and new types of components in the fast path.